### ISSUE TITLE
"Stuck client" may not be detected in case of non-open socket

### ISSUE THREAD

**Issue by eolivelli** [diennea/blazingcache#148]
We had a case in which a client was partitioned from the network but pending "invalidate key" did not timeout.

Looking at 
https://github.com/diennea/blazingcache/blob/master/blazingcache-core/src/main/java/blazingcache/server/CacheServerEndpoint.java
and
https://github.com/diennea/blazingcache/blob/master/blazingcache-core/src/main/java/blazingcache/network/netty/NettyChannel.java

the channelIdle notification is sent only to "valid" channels, this notification handles stuck messages.

*Maybe* the socket was not reported as 'open' so the callback was not called, leading to the problem

```
 void processIdleConnection() {
        Channel _channel = channel;
        if (_channel != null && _channel.isValid()) {
            _channel.channelIdle();
        }
    }
```

**Comment by eolivelli** [comment]
It looks like that we got  into this problem again in production.
A machine was in bad state (needed OS forced reboot) and the server failed to detect it as stuck.

**Comment by eolivelli** [comment]
I am working on this issue now, expect to see it released with 2.3.0, within a couple of weeks
